### PR TITLE
[10.0][ADD] module account_type_inactive

### PR DESCRIPTION
**Pull request by grindtildeath**
Account Type Inactive
=====================

This module extends the functionality of account module to allow you to flag
account types as active or inactive.

### CODE REVIEW

**Comment by fclementic2c** [comment]
@grindtildeath runbot please :)

File: account_type_inactive/README.rst [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/82/10.0
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues
+<https://github.com/OCA/{project_repo}/issues>`_. In case of trouble, please
```
Comment (damdam-s) [inline account_type_inactive/README.rst]: s/{project_repo}/account-financial-tools/

**Comment by sbidoul** [comment]
I remember we've had problems with accounts flagged as inactive, especially for reports that would select accounts before selecting move lines, ending up ignoring move lines on accounts that were inactivated.

In particular mis_builder will have such problems, because it selects accounts first then read_group on move lines filtered on selected accounts. I'm pretty sure other modules will fail in subtle ways with inactive accounts.

File: account_type_inactive/README.rst [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+Account Type Inactive
+=====================
+
+This module extends the functionality of account module to allow you to flag
+account types as active or inactive.
+
+Installation
+============
+
+Just install it. Existing account types will be flagged as active.
```
Comment (elicoidal) [inline account_type_inactive/README.rst]: "Installing the module will keep existing account types flagged as active."

File: account_type_inactive/views/account_type.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="view_account_type_form" model="ir.ui.view">
+        <field name="name">account.account.type.form.inherit</field>
+        <field name="model">account.account.type</field>
+        <field name="inherit_id" ref="account.view_account_type_form" />
+        <field name="arch" type="xml">
+            <field name="type" position="after">
+                <field name="active" />
```
Comment (elicoidal) [inline account_type_inactive/views/account_type.xml:9]: Shouldnot inactive type not displayed by default (default Odoo behavior)?

**Review by elicoidal — APPROVED** [review]
small details

**Comment by fclementic2c** [comment]
I get your point Stephane and It's annoying.

In fact what I really need here is to unactivate odoo user_type so I can work on customized user_type and base financial reports (B/S and P&L mostly) on user_type instead of range of account.

2 possibilities:
- Either we do not put such a module into OCA branch at all
- Or we consider this is advanced settings (the menu does even exist), we add a warning on the readme.
I am divided.
What would you prefer @elicoidal @sbidoul ? I'll follow your vote.

**Comment by sbidoul** [comment]
Perhaps you could prevent the deactivation of account types that are associated with accounts which have move lines?

**Comment by fclementic2c** [comment]
@sbidoul Ok let's for that @grindtildeath

**Comment by elicoidal** [comment]
> Perhaps you could prevent the deactivation of account types that are associated with accounts which have move lines?

:+1:

File: account_type_inactive/views/account_type.xml (hunk 3 above)
Comment (grindtildeath) [reply account_type_inactive/views/account_type.xml:9]: The field type in this view is the 'Internal type' Selection field and is only used to display the active checkbox below it on the form view.
Account types marked as inactive won't be displayed on the tree view or Many2one selectors (default odoo behaviour) anyway.

**Comment by grindtildeath** [comment]
@fclementic2c @sbidoul @elicoidal 
I added a constraint so that it's not possible to inactive an account type, should this account type be used on an account.account.
The problem with the check on account with move lines generated (as proposed here), is that it's still possible to inactivate an account type which is used on an account, and then create a move.line on this account. 
As it looks safer to restrict the inactivation here, that's what I did at the moment.
Another possibility would be to mark every account account as deprecated if every accounts using this type don't have move lines, but I don't think it's a good solution neither.
I'm open to your suggestions.

**Comment by sbidoul** [comment]
@grindtildeath good enough for me!

**Review by sbidoul — APPROVED** [review]

**Comment by elicoidal** [comment]
:+1:

**Review by yvaucher — APPROVED** [review]

**Pull request merged**